### ISSUE TITLE
[Feature request] Make 3 vertical column layout

### ISSUE THREAD

**Issue by dauphine-dev** [dauphine-dev/drop-feeds#64]
Make 3 vertical column layout :
 - one for feeds (already exist)
 - another vertical column for article-titles (already exist)
 - and the third one for article-content (has to be added)

_Feature requested in user review on AMO ([here](https://addons.mozilla.org/fr/firefox/addon/drop-feeds/reviews/1222108/))_

**Comment by dauphine-dev** [comment]
@atomGit I have close the other [issue]() and merge it with the current one.
We can continue to speak in this thread.

*From the other issue:*

------

> given that DF displays feed content in the main browser view port, i wonder why it's useful to display the feed list in the sidebar?
> 
> this may be beyond the goals of the developer, but personally i am very much seeking a WebExt feed reader similar to the popular and well liked Newsfox - a traditional 3 pane display with feeds on the left, headlines on the top-right and article content on the bottom-right
> 
> in my opinion this is a very efficient way of reading feeds in that it avoids a lot of extra scrolling through content you don't care about because feed content is only displayed when a headline is clicked
> 
> also Newsfox was able to load all headlines from a feed group (folder) and i miss the site icons as well

------

> Hi @atomGit
> Welcome !
> You are the one has post this [review](https://addons.mozilla.org/fr/firefox/addon/drop-feeds/reviews/1222108/) ?
> 
> About the 3rd layout a issue is already open [here](https://github.com/dauphine-dev/drop-feeds/issues/64).
> 
> What ever I am working on the 3rd panel. I have tried to install Newsfox, but it is not any more compatible with FF57+, and it has failed to be installed on my FF56 (I haven't investigate why...)
> 
> See commits on 0.9.1 branch, you can event play with the development version by cloning this repo and open it as a temporary add-on, may be it better if you do that on a secondary Firefox session or on other Firefox installation (like Firefox Developer).
> As soon I get some thing acceptable (I am close...) I will make an new alpha version (alpha4) [here](https://github.com/dauphine-dev/drop-feeds/releases) with the 3rd panel, then it will be easier to install/test/use. Then you will can tell to me if it is matches with you expectations.

> given that DF displays feed content in the main browser view port, i wonder why it's useful to display the feed list in the sidebar?
> 
> this may be beyond the goals of the developer, but personally i am very much seeking a WebExt feed reader similar to the popular and well liked Newsfox - a traditional 3 pane display with feeds on the left, headlines on the top-right and article content on the bottom-right
> 
> in my opinion this is a very efficient way of reading feeds in that it avoids a lot of extra scrolling through content you don't care about because feed content is only displayed when a headline is clicked
> 
> also Newsfox was able to load all headlines from a feed group (folder) and i miss the site icons as well

**Comment by dauphine-dev** [comment]
@atomGit
Drop feeds 0.9.1alpha4 is now available [here](https://github.com/dauphine-dev/drop-feeds/releases)
Please tell me it the 3rd panel work as you expect.

**Comment by atomGit** [comment]
looks like one of my posts got lost - in answer to your question...

> You are the one has post this review ?

no, but it is interesting that someone else commented on the 3-pane layout of Newsfox

> Drop feeds 0.9.1alpha4 is now available here
> Please tell me it the 3rd panel work as you expect.

this is good!

the basic components are in place, but i ran into some bugs and have some suggestions...

Newsfox (NF) did not make use of the sidebar at all and, personally, i feel that the sidebar is too small to display content from all 3 panels - NF used only the main viewport (where webpages are displayed) and though the layout could be changed, i think the most efficient layout was the traditional 3 pane layout like this...

```
--------------------------------------
|            |
|  FEED      | ARTICLE HEADLINES
|  LIST      |----------------------
|            |
|            | ARTICLE CONTENT
|            |
--------------------------------------
```

so the feed list panel contained what your 1st panel does now - the folders and feed titles

if you clicked on a feed, all the article headlines from that feed would be displayed in the article headlines panel (what is in your 2nd panel)

if you clicked on a headline, the article content was displayed in the bottom-right panel

the article content panel, as above, also contained the article title and if you clicked on the title, the article (web page) would open in a new tab - as i recall, you could also open the article in a new tab by double clicking? or middle-clicking? the article headline in the article headlines pane

i think putting everything in the sidebar might make sense for those using a big display, but for people like me who use a laptop, the sidebar is just too small to hold that much content (i subscribe to a LOT of feeds) - of course the nice thing about the sidebar is that you never have to change tabs, but that benefit is debatable

some bugs i noticed...

settings -> themes: if i changed the theme from 'legacy', the entire sidebar went blank except for the top toolbar - in some cases the background color went solid red - changing it back to 'legacy' did not resolve the problem - i had to clean out the extension data and reinstall to get things working again - this may be because i installed 0.9.1alpha4 over the last version perhaps

even when the top panel in the sidebar was working, the bottom 2 panels did not appear at all until i toggled the 'feed item list' setting in settings -> items

in the bottom panel, there's no scroll bars in the article content - the content just gets cut off

**Comment by atomGit** [comment]
also, i see i kind of hijacked the OP's thread :)  -sorry

if the OP is open to displaying the feed stuff in the main viewport as i'm suggesting, then it should be trivial (CSS) to arrange the 3 panes like he/she wants (all vertical) or as i suggested above - i think having the content in the web page viewport will provide you more freedom layout wise

**Comment by dauphine-dev** [comment]
Thank you for your feed back.

Yeah there some bugs it is because it is an aplha version, I fix only big ones and I keep small ones for beta phase when feature development  will be completed. But yes reports then because I can have missed some ones ...

My target/model is Sage/Sage+ then every things happen in the side bar with a list of "Headlines+Article" in the tab page.
We call that "(headlines+content) list"


But yes I will try to think about :

```
--------------------------------------
|            |
|  FEED      | ARTICLE HEADLINES
|  LIST      |----------------------
|            |
|            | ARTICLE CONTENT
|            |
--------------------------------------
```
We call that  "headline list + article"

Drop Feed haven't be think to that, I will see what I can do, but if I have to change most of parts of the Drop Feeds code, I will not do it  (I haven't the bravery  for that....)
But I will look may be there is a way to do it easily

**Comment by dauphine-dev** [comment]
@atomGit 
I have made a new render template called "Two panels (Fox Memory)", and a new theme called "Fox Memory", with these settings you should get some thing look like FoxNews ("look like" but not "same features", Drop Feeds is still targeting Sage/Sage++ features)

Please try it, and tell me what you think about ?

**Comment by atomGit** [comment]
i appreciate your work and willingness to satisfy feature requests, but i think that Drop Feeds will not work for me - displaying feed lists and article headlines in the sidebar is too crowded for those with smaller displays and a large number of feeds

i am sure others will appreciate the new features

**Comment by dauphine-dev** [comment]
No problems :-)

I hope the one who has post the [review](https://addons.mozilla.org/fr/firefox/addon/drop-feeds/reviews/1222108/), will like it.

Now with "Fox Memory" render template, you should get a larger view more adapted for small displays, but I guess there are some extensions or app better designed for small screens than Drop Feeds.

**Comment by dauphine-dev** [comment]
Third layout is available in 0.9.1

**Comment by atomGit** [comment]
hi @dauphine-dev 

i'm trying the new layout now and i like the 3 panes - i have some feedback and, if you want, i'll open new tickets for each of these that you want to address, but i just wanted to list some stuff and get your thoughts first...

1) for any place where a context menu can be displayed (right-click), it would be nice if pressing the Esc key would close it (i don't know if that's possible with the WebExt API)
2) theme -> render tab template : the option 'two columns' should be renamed 'two panes' or 'two rows' since one panel is above the other, not side-by-side
3) when 'two columns' is selected, feed headlines are displayed both in the sidebar and the upper pane in the reader tab - this is redundant? would be better to remove the headlines panel in the sidebar and use all that space for displaying the list of feeds
4) when clicking a feed folder (a group of feeds), it would be super-appreciated if all of the headlines for all of the feeds in that group appeared in the upper pane of the reading tab - it takes too long to click on every feed to view all the headlines when one has a lot of feeds :)
5) it looks like the dark themes don't work very well (at least not on Linux with the default dark theme for FF) - i will be happy to code a new dark theme if you want

by the way, i very much like the settings for allowed html elements

**Comment by dauphine-dev** [comment]
Hi @atomGit 

> 1) for any place where a context menu can be displayed (right-click), it would be nice if pressing the Esc key would close it (i don't know if that's possible with the WebExt A

It is already done on 0.9.2 version (coming soon in production)

> 2) theme -> render tab template : the option 'two columns' should be renamed 'two panes' or 'two rows' since one panel is above the other, not side-by-side

I am not sure to understood, "One column (Sage Single Column)" has only one column, then no problem, "Two columns (Legacy) has two columns", no ? and "Two panels (Fox Memory)" has 2 rows (or panels), no ?

> 3) when 'two columns' is selected, feed headlines are displayed both in the sidebar and
>  the upper pane in the reader tab - this is redundant? would be better to remove the headlines  panel
>  in the sidebar and use all that space for displaying the list of feeds
> 

When you say "two columns'", I guess speak about "Two panels (Fox Memory)", if not then I haven't understood you question. If yes, then "Two panels (Fox Memory)", it is clone a the render view of "FoxNews reader"  another RSS reader on FF 56 has died with FF57, it has been made for "FoxNews reader" nostalgic. I will complete my answer in point 5...

> 4) when clicking a feed folder (a group of feeds), it would be super-appreciated if all of the headlines for all of the feeds in that group appeared in the upper pane of the reading tab - it takes too long to click on every feed to view all the headlines when one has a lot of feeds :)

I guess, you are still speaking about "Two panels (Fox Memory)" render view. It is only a render view, it is independent of the sidebar the are no interaction between both, then I won't do that. And the same thing it is clone of the render view of "FoxNews reader"  for nostalgic, and "FoxNews reader" had the items list in top panel, not the feeds list.

> 5) it looks like the dark themes don't work very well (at least not on Linux with the default dark theme for FF) - i will be happy to code a new dark theme if you want

I am GNU/Linux user too (Mint Cinnamon), but I am bad to making theme, yes I will happy if you can make a good dark theme, it is a CCS style sheet, (there are 2 in fact, one for the positioning, and one for colors, then you can apply a color sheet, to different positioning), and about point 3 you can do your own render template it is XSLT file. In a new version (0.9.3 I guess) I will make a theme editor.

> by  the way, i very much like the settings for allowed html elements

It was a feature of Sage++ (I don't remember if it was in Sage), then naturally it is in Drop Feeds (I try to have in Drop Feeds same features than in Sage/Sage++)

**Comment by atomGit** [comment]
> I am not sure to understood, "One column (Sage Single Column)" has only one column ...

yes, you're right - i missed the checkbox for 'feed item list' in the 'items' tab

regarding point 4, i'm not sure we are understanding each other...

> when clicking a feed folder (a group of feeds), it would be super-appreciated if all of the headlines for all of the feeds in that group appeared in the upper pane of the reading tab

so what i mean is: if you use the 'two panels' render template and you click on a feed item in the sidebar, a list of headlines for that feed is displayed in the upper panel of the rendering tab - this is perfect - however what i am asking for is that, when you click on a **folder** in the sidebar, all of the headlines for that folder should be displayed in the rendering tab ... and if you clicked on the root folder ("Drop Feeds" by default), then all of the headlines for every feed group would be displayed

this is sort of like your 'unified feed' option, but you would still have the 2 panels - one with headlines and one with the article content (both in the render tab)

BTW, when i tried the 'open updated as a unified feed', nothing was displayed in the render tab - just the group name as the title and "Unified Feed" as the content

**Comment by dauphine-dev** [comment]
That "Unified Feed" do.

Wait, I have just checked "Unified Feed" with "Two panels (Fox Memory)", and it doesn't work (it works with other render templates)

Then I will look to fix that in 0.9.2

(You can start to play with 0.9.2rc here: https://github.com/dauphine-dev/drop-feeds/releases)

**Comment by dauphine-dev** [comment]
"Unified Feeds" are fixed in 0.9.2rc2.
I think It will be the final one then I should be make it available on AMO this evening.

But please play with 0.9.2rc2 and tell me if you see any issues.
